In [7]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [21]:
df = pd.read_csv('./tool_wear_detection/train.csv')

In [22]:
df.describe(include='all')

No material   feedrate  clamp_pressure tool_condition  \
count   18.000000       18  18.000000       18.000000             18   
unique        NaN        1        NaN             NaN              2   
top           NaN      wax        NaN             NaN           worn   
freq          NaN       18        NaN             NaN             10   
mean     9.500000      NaN   9.722222        3.416667            NaN   
std      5.338539      NaN   7.282578        0.624264            NaN   
min      1.000000      NaN   3.000000        2.500000            NaN   
25%      5.250000      NaN   3.000000        3.000000            NaN   
50%      9.500000      NaN   6.000000        3.500000            NaN   
75%     13.750000      NaN  18.750000        4.000000            NaN   
max     18.000000      NaN  20.000000        4.000000            NaN   

       machining_finalized passed_visual_inspection  
count                   18                       14  
unique                   2                        2  
top                    yes                      yes  
freq                    14                       10  
mean                   NaN                      NaN  
std                    NaN                      NaN  
min                    NaN                      NaN  
25%                    NaN                      NaN  
50%                    NaN                      NaN  
75%                    NaN                      NaN  
max                    NaN                      NaN

In [24]:
df.head(50)

No material  feedrate  clamp_pressure tool_condition machining_finalized  \
0    1      wax         6             4.0         unworn                 yes   
1    2      wax        20             4.0         unworn                 yes   
2    3      wax         6             3.0         unworn                 yes   
3    4      wax         6             2.5         unworn                  no   
4    5      wax        20             3.0         unworn                  no   
5    6      wax         6             4.0           worn                 yes   
6    7      wax        20             4.0           worn                  no   
7    8      wax        20             4.0           worn                 yes   
8    9      wax        15             4.0           worn                 yes   
9   10      wax        12             4.0           worn                 yes   
10  11      wax         3             4.0         unworn                 yes   
11  12      wax         3             3.0         unworn                 yes   
12  13      wax         3             4.0           worn                 yes   
13  14      wax         3             3.0           worn                 yes   
14  15      wax         6             3.0           worn                 yes   
15  16      wax        20             3.0           worn                  no   
16  17      wax         3             2.5         unworn                 yes   
17  18      wax         3             2.5           worn                 yes   

   passed_visual_inspection  
0                       yes  
1                       yes  
2                       yes  
3                       NaN  
4                       NaN  
5                        no  
6                       NaN  
7                        no  
8                        no  
9                        no  
10                      yes  
11                      yes  
12                      yes  
13                      yes  
14                      yes  
15                      NaN  
16                      yes  
17                      yes

In [25]:
df2 = df = pd.read_csv('./tool_wear_detection/experiment_01.csv')

In [26]:
df2.head()

X1_ActualPosition  X1_ActualVelocity  X1_ActualAcceleration  \
0              198.0                0.0                   0.00   
1              198.0              -10.8                -350.00   
2              196.0              -17.8                  -6.25   
3              194.0              -18.0                   0.00   
4              193.0              -17.9                 -18.80   

   X1_CommandPosition  X1_CommandVelocity  X1_CommandAcceleration  \
0               198.0                 0.0                0.000000   
1               198.0               -13.6             -358.000000   
2               196.0               -17.9               -0.000095   
3               194.0               -17.9               -0.000095   
4               192.0               -17.9                0.000095   

   X1_CurrentFeedback  X1_DCBusVoltage  X1_OutputCurrent  X1_OutputVoltage  \
0                0.18           0.0207             329.0              2.77   
1              -10.90           0.1860             328.0             23.30   
2               -8.59           0.1400             328.0             30.60   
3               -6.11           0.1300             327.0             30.30   
4               -5.70           0.1140             328.0             30.50   

         ...          S1_CurrentFeedback  S1_DCBusVoltage  S1_OutputCurrent  \
0        ...                       0.524     2.740000e-19             329.0   
1        ...                      -0.288     2.740000e-19             328.0   
2        ...                       0.524     2.740000e-19             328.0   
3        ...                      -0.180     2.740000e-19             328.0   
4        ...                       0.226     2.740000e-19             328.0   

   S1_OutputVoltage  S1_OutputPower  S1_SystemInertia  \
0               0.0    6.960000e-07              12.0   
1               0.0   -5.270000e-07              12.0   
2               0.0    9.100000e-07              12.0   
3               0.0    1.070000e-06              12.0   
4               0.0    1.960000e-06              12.0   

   M1_CURRENT_PROGRAM_NUMBER  M1_sequence_number  M1_CURRENT_FEEDRATE  \
0                        1.0                 0.0                 50.0   
1                        1.0                 4.0                 50.0   
2                        1.0                 7.0                 50.0   
3                        1.0                 7.0                 50.0   
4                        1.0                 7.0                 50.0   

   Machining_Process  
0           Starting  
1               Prep  
2               Prep  
3               Prep  
4               Prep  

[5 rows x 48 columns]

## Final Project Description
----

The goal of the final project for me is to be able to predict the following characteristics:

#### Tool wear detection: 
Supervised binary classification could be performed for identification of worn and unworn cutting tools. Eight experiments were run with an unworn tool while ten were run with a worn tool (see tool_condition column for indication).

#### Detection of inadequate clamping: 
The data could be used to detect when a workpiece is not being held in the vise with sufficient pressure to pass visual inspection (see passed_visual_inspection column for indication of visual flaws). Experiments were run with pressures of 2.5, 3.0, and 4.0 bar. The data could also be used for detecting when conditions are critical enough to prevent the machining operation from completing (see machining_completed column for indication of when machining was preemptively stopped due to safety concerns).